utils/frame/try-runtime/cli/src/lib.rs

```diff
@@ -114,12 +126,18 @@ pub struct TryRuntimeCmd {
 /// The source of runtime state to try operations against.
 #[derive(Debug, Clone, structopt::StructOpt)]
 pub enum State {
-	/// Use a state snapshot as the source of runtime state. NOTE: for the offchain-worker command this
-	/// is only partially supported at the moment and you must have a relevant archive node exposed on
-	/// localhost:9944 in order to query the block header.
-	// TODO https://github.com/paritytech/substrate/issues/9027
+	/// Use a state snapshot as the source of runtime state. NOTE: for the offchain-worker and
+	/// execute-block command this is only partially supported and requires a archive node url.
 	Snap {
+		#[structopt(short, long)]
 		snapshot_path: PathBuf,
+		/// The url to connect to for fetching data not easily retrievable from the snapshot.
+		/// NOTE: This is not necessary for `on-runtime-upgrade`
+		// TODO This is a temporary hack; the url is used just to get the header/block. We should try
+		// and get that out of state, OR allow the user to feed in a header/block via file.
+		// https://github.com/paritytech/substrate/issues/9027
+		#[structopt(default_value = "ws://localhost:9944", parse(try_from_str = parse::url))]
```

Comment:
```suggestion
		#[structopt(long, default_value = "ws://localhost:9944", parse(try_from_str = parse::url))]
```

IMO we get a nicer cli when things are flags by default instead of positional args.

Reply:
yeah agreed; changing the other url arg as well to a flag for symmetry.